### DISCUSSION TITLE
Giraffe Communication

### DISCUSSION THREAD

**Discussion by ziproot** [ESMerchantCampaign/merchant-campaign#4] [Announcements]
Since Giraffe doesn’t have Discord, he can comment here.

**Comment by TheGiraffe3** [comment]
Or rather, since Giraffe is locked out of Discord, he can communicate here.

**Reply by SpearDane** [reply]
just found this... sorry for being late to the party.

Will you ever not be locked out? Or is that just your life now?

**Reply by TheGiraffe3** [reply]
It's unclear. The current problem is that click.discord.com isn't working. I'm theorizing that's because of Saudi Arabia's firewall, so next time I travel I might be able to get in.
We'll see.

**Reply by ziproot** [reply]
Is it possible for you to create a [Matrix](https://matrix.org/try-matrix/) account? If so, you can join the following room, and I can invite everyone to a Discord server that is bridged to it: https://matrix.to/#/!bWVAHKHnFDzdXeKlSp:tchncs.de?via=tchncs.de&via=t2bot.io
EDIT: I have deleted the room since Giraffe has found a way back into Discord, so the link will not work

**Reply by TheGiraffe3** [reply]
I've rejoined Discord. We'll see how this goes.
I can't verify my account, so I may get booted off again (and I don't get access to Swizzle 6).

**Reply by TheGiraffe3** [reply]
Never mind; I was told I needed to verify, which isn't possible right now.
I'll look into Matrix later.


I have opened a support ticket asking about it, so potentially I'll be able to get in that way.

**Reply by TheGiraffe3** [reply]
Alright, problem solved!
I'll close this discussion now as it was opened because of Discord lack.

Will note that I don't currently have access to #core-contributing though.

**Comment by TheGiraffe3** [comment]
Thoughts on rewards for beating the campaign:
Jump Drive: every line gets this.
Perhaps access to a superpowered warship that the Mercantile has been developing; but it's slow, ungainly, and has many required crew but few bunks.

**Reply by ziproot** [reply]
Dane on Discord thinks the main reward (other than the obligatory Jump Drive) could be a type of super-Argosy

**Reply by mOctave** [reply]
The super warship idea is kind of cool, but it doesn't really fit for the non-mercenaries. For the pacifists/merchants, what about better/more cost-efficient freighters or transports, or something along those lines?

**Reply by TheGiraffe3** [reply]
~~guess that reveals my playstyle~~
Good point. I'll note though that we've already got a _massive_ amount of human freighters (though I suppose it's the same with human warships).
Maybe an improved cargo expansion for merchants, and an improved outfits expansion for privateers?

**Reply by SpearDane** [reply]
Super-Argosy was just an example of an idea. I do think a ship should be the reward. I brought up a Super-Argosy as one idea. I also brought up the [Cambrian](https://evn.fandom.com/wiki/Cambrian) and and [Pirate Carrier](https://evn.fandom.com/wiki/Unrelenting) from EVN as other possible directions to go in. 

If it's larger, like the Cambrian/Pirate Carrier it could be easier to have branch specific kits.

**Comment by TheGiraffe3** [comment]
> README: mOctave is working on bounty hunter jobs

As in jobs to hunt pirates or jobs where pirates hunt you?

**Reply by ziproot** [reply]
Jobs to hunt pirates

**Comment by TheGiraffe3** [comment]
Could you turn on the "Update Branch" button? Manually updating is much more difficult then hitting a button.
~~it's actually easier at present to make conflicts~~

**Reply by ziproot** [reply]
How do I do that?

**Reply by TheGiraffe3** [reply]
Settings/General and then under "Pull Requests" there's a... screenshot:
![Screen Shot 2024-12-03 at 7 59 49 AM](https://github.com/user-attachments/assets/bce58057-9c19-4c5d-9aa3-1a85a284040b)
Check "Always suggest updating pull request branches"

**Reply by ziproot** [reply]
Done

**Reply by TheGiraffe3** [reply]
Thanks.

**Comment by TheGiraffe3** [comment]
Thoughts on my updated Flowchart? Or should we work on other parts of the campaign first?
https://github.com/ziproot/merchant-campaign/tree/TheGiraffe3-flowchart-update

**Comment by TheGiraffe3** [comment]
A while back on Discord, SpearDane and I were talking about how a potential merchant campaign could involve Hroar's Haulers from the Deep.
Has there been further discussion about that?

**Reply by SpearDane** [reply]
Why do you think I am here!

I'll find a way to involve them, and the Transport Union.

**Reply by roadrunner56** [reply]
I'd love to see Hroar's Haulers again!

**Reply by SpearDane** [reply]
I am also working on that too...

**Comment by ziproot** [comment]
![image](https://github.com/user-attachments/assets/18acbbc9-bc9f-4669-98a4-70cfebf8c060)

**Comment by ziproot** [comment]
Self-hosted rulesets should work now (as long as I am running my computer)

**Reply by TheGiraffe3** [reply]
As long as you don't have to pay, sounds good.

**Comment by ziproot** [comment]
From Dane on Discord:
Merchant something like this?

Job Board missions
General Opener/introduction to RM
General Missions
Big event Mission related to Civil War Timeline

Pacifist Mission string
Pacifist Job board missions 2.0
Mecenary Mission string
Mercenary Job board missions 2.0
Merchant branch opener
Merchant missions
Merchant Job Board missions 2.0
Big General Mission related to Civil War with serious choices

Pacifist Mission string
Pacifist job board missions 3.0
Mercenary Mission string
Mercenary job board missions 3.0
Merchant missions
Merchant Job Board missions 3.0
Big Late game event!
End of Merchant campaign 

After-campaign job board missions.

**Reply by TheGiraffe3** [reply]
Would we want to add this to a `timeline.txt`?

Related: there's the [Core Mining Inc.](https://github.com/DJF113/Core-Mining-Inc) plugin from a while ago. The original author semi-abandoned it and I was wondering about cleaning it up and PRing it to vanilla ES. Would there be interest in including that as part of the Merchant Campaign (and thus I shouldn't PR it get)?

(also, edited your comment to fix instances of merhcant)

**Reply by SpearDane** [reply]
Yes. I think that's be a great idea to add that into the campaign as a side mission.

So, I think the first section of the Merchant Campaign should be available to all players - regardless of what campaign they do. So one thing to consider is if this CMI plugin is part of the content anyone can do - or the content that gated behind committing to the Merchant Campaign. 

If there are any other abandoned plug ins or unfinished PR's that we could cannibalize.. lets do it. There's a lot of ground to cover in a campaign!

**Reply by TheGiraffe3** [reply]
Well, it's very long for a "anyone can do"; it spans four files. But after the first file the CMI eliminates Stormhold as a pirate base and takes it over, which could potentially have serious trouble with the Korath campaign, albeit that's not in the game yet.

As for unfinished PRs we can cannibalize, it looks like the only one that
- wasn't closed for good (e.g. won't fit vanilla)
and
- isn't going to be picked up again (by bene)
is the Republic Campaign Project, which we can't use, and the Build-A-Mine mission string. 
The problem with that one is that it's hardly written at all and alexrovw got a job prohibiting him from contributing to open-source projects. So it could be a guideline, so to speak, but it might be hard to cannibalize.
https://github.com/endless-sky/endless-sky/pull/9637

**Reply by TheGiraffe3** [reply]
As for plugins, it looks like there's not really any which add a merchant campaign. The only one that might have what we're looking for is "Civil War" which apparently takes the data from a hard fork of ES.
https://github.com/Hecter94/EndlessSky-PluginArchive/tree/main/Working/Civil%20War/data

**Reply by SpearDane** [reply]
Ah, so that's what happened to alexrovw. Shame. 
Alright, well if CMI is all we get - still better than nothing! Let's jam that bad-boy in!

**Reply by TheGiraffe3** [reply]
Yep, I asked if I could take over the Charles and Donna followup (which later got closed) and was told then.

**Reply by SpearDane** [reply]
![RMCampStory](https://github.com/user-attachments/assets/4552e120-9d41-4391-9b8a-6afd2a721fe1)
 
Updated graphic of the above idea. This is different from the flowchart in that it is not charting a relationship with the larger FW/Civil war story. It's just showing how the branches could function. I think it's nice if in the prologue you get a taste of all three walks of life for a merchant. 

Perhaps there is a prologue version of the CMI stuff as well?

Then the player is given the offer to join the RM. If they agree they work into the next phase. Here they do even more with all three branches. And after doing a bit with them they finally get a choice to either keep going down that branch, or return to the Merchant branch. As I've mentioned on the discord, I think we shouldn't build out either Pacifist or Mercenary branch until after we have the Merchant branch done - so that we don't stretch ourselves too thin and can at least ensure one portion of the campaign is finished. 

I think we may be able to still include a few Pacifist and Mercenary missions in later stages of the Merchant branch, but they might not be as connected to the either of those branches at that point.

**Reply by TheGiraffe3** [reply]
After a while the CMI gets a lot of pirate fighting. Perhaps a few of the first pacifist missions are transporting Michael (the main character) and then the Mercenary storyline is mainly CMI stuff?

**Reply by SpearDane** [reply]
Octave has some pretty specific ideas about the Mercenary line. But we could have the CMI grow out of a Mercenary mission that everyone gets.

**Reply by TheGiraffe3** [reply]
I'd be interested in hearing those ideas.

That would be possible. But in the initial meeting with Michael Washington he doesn't seem to indicate that he's got any intentions of starting a war; and in fact I don't think violence would be necessary without the pirate invasion.
The CMI is in a weird place between lots of pirate combat and lots of transport. Almost makes me wonder if it should be part of the Merchant storyline at all - or another branch - or the Merchant Prologue before the split, though that seems unlikely.

**Reply by TheGiraffe3** [reply]
> As I've mentioned on the discord, I think we shouldn't build out either Pacifist or Mercenary branch until after we have the Merchant branch done - so that we don't stretch ourselves too thin and can at least ensure one portion of the campaign is finished.

Alternatively, we could "assign" certain people certain branches if they aren't interested in, for example, working on huge battles.

**Reply by SpearDane** [reply]
Well, one idea for CMI was have it be like the wolfpack missions in the FW campaign. They start separately, and are somewhat contained in their own little bubble - but ramifications of doing them can appear in the other branches. 

As for what Octave wrote for the mercenary campaign: 

mOctave — 12/2/2024 5:25 PM
_Okay, ideas for the mercenary campaign:
Prologue - Piracy increases with the outbreak of war. You take bounty hunting jobs against pirates. Taking enough of these leads to an early spaceport mission where you're not committed to any one branch yet, but are called to fight a pirate fleet that is masquerading as Free Worlds ships. At the end of this mission chain you are given the opportunity to join the Mercantile.
Early - More anti-piracy, escort defense, etc. missions. At one point, you have to take out a Navy warship and flee Republic space. After being in exile in <insert area here> for a little while, you receive a notice exonerating you of your crimes, and get called up before the Navy. It appears that the ship you destroyed was some sort of Giftbringer equivalent.
Mid-Late - Investigation into this pirate gang that is causing this impersonation, eventually leading to the Alphas, or choosing to profit based on the war and drag it out. Both paths eventually lead to you fighting in the Pug war—in the first you are called to fight as a moral champion with a powerful fighting force; in the second you are paid to fight._

In the chart above, the choice break for the Mercenary campaign, would presumably be destroying the Navy ship in Octave's write up. The non-Mercenary branch pilot does some recon and locates it, but then then actually going to destroy it requires a commitment to the mercenary branch. Otherwise they just tell the pilot, "Thanks for the help - we've got it from here." And the pilot goes back to the merchant missions.

**Reply by TheGiraffe3** [reply]
That looks good.
If CMI isn't going to be a part of the main storyline, the main storyline should definitely accommodate it, because it makes rather extreme changes.
And I just realized that HR invades Stormhold...
Alright, so to tie in with the bounty approach #20 adds new conditions to track how many pirates you've killed.

I don't see a pacifist outline there though?

**Reply by ziproot** [reply]
I don’t think it is necessary to use **all** of the plugin, but I still think it could be condensed and included as a side plot. Therefore, parts that make extreme changes (such as the invasion of Stormhold) can be skipped over.

PS: Next week is my final exam week (from Wednesday to the Tuesday of the week after) so I will not only not be available, but my computer will be shut down during exams. If you want, you can use GitHub’s runners instead of mine for that time since I don’t think 2,000 minutes will be used up in seven days. My runners should still work most of the time from now until Wednesday. This requires changing the fields for checks/spelling to ubuntu-latest instead of self-hosted, and the fields for the data parser from self-hosted to windows-2022. It also requires changing `ziproot/endless-sky-test-parse@main` to `warp-core/endless-sky-test-parse@master` since my script only works for my machine.

**Comment by TheGiraffe3** [comment]
Is the Merchant Campaign "public knowledge" as in "people know it a being worked on"?

**Reply by SpearDane** [reply]
Core and Devs know its being worked on. We haven't spoke about it in the more public channels on discord though - to my knowledge. It's not a secret though.

**Reply by TheGiraffe3** [reply]
So I suppose it's a bad idea to say in the CMI discussion I posted "this will be in the merchant campaign"?

**Reply by SpearDane** [reply]
That's ultimately up to Zip, but I think it's fine.

**Reply by ziproot** [reply]
> Core and Devs know its being worked on. We haven't spoke about it in the more public channels on discord though - to my knowledge. It's not a secret though.

This is technically inaccurate: I have also posted this to swizzle 6 since I was not sure if some of the people there not on ESC wanted to help, but swizzle 6 still has less people on it than ESC main. I think that you can post that it will be in the merchant campaign since that is similar to swizzle 6 in that most people won’t see it.

**Comment by TheGiraffe3** [comment]
It occurs to me, would we want to move this repository and the [CMI one](https://github.com/TheGiraffe3/Core-Mining-Inc) to an organization?
I believe we'd be able to do some of the currently-restricted things in a private organization repository, and it allows for much more customizability in access rights and matters of that sort.

**Reply by TheGiraffe3** [reply]
I guess the chief advantage would be using GitHub runners so that checks will always run immediately 😉

**Reply by ziproot** [reply]
Is it possible to do so without destroying my account?

EDIT: It looks like it is possible, but I do not gain any additional minutes, nor the ability to add rulesets to private repos, unless I also pay $4 a month.

**Reply by TheGiraffe3** [reply]
I believe if you "transfer" the repository nothing will happen to you; the organization "pays" for the Actions minutes.
No rulesets though, apparently.

**Reply by ziproot** [reply]
See, this is the problem
![image](https://github.com/user-attachments/assets/4644fa63-3498-4f17-a096-0627b7c79e45)

The organization still has 2,000 minutes, that doesn’t change.

**Reply by mOctave** [reply]
We would be able to split ourselves up into teams (create a review team or something like that, maybe me, Giraffe, and bene?).

**Reply by TheGiraffe3** [reply]
That, and I was assuming the reason you were using `self-hosted` here was so that you could use GHActions minutes for other projects.
I've been contributing typo fixes to several projects, and because Minutes reset after every month I've never used over a thousand.
I believe for two repositories active on an organization, not over a thousand would be used, and even if we use 1500 nobody's paying for it.

**Reply by TheGiraffe3** [reply]
Well, we're at 119 minutes out of 2000. I'm estimating that's because of the "PR flood" having to do with the repository change; though we may get closer then my original estimate of 1500.
It's likely the data files check that takes the time.

**Reply by ziproot** [reply]
Yeah, data files uses four minutes (two minutes to run the check × 2 because it’s windows)

**Reply by TheGiraffe3** [reply]
Turns out we also get to use draft PRs.

**Comment by mOctave** [comment]
Okay, here's what I'm thinking for the Mercenary prologue, in more detail:
```
Mercenary Prologue 1: FL is introduced to a couple of captains who are running supplies for a mining operating (à la CMI Taxi). These captains enlist FL's help as an escort, since they're worried about increased pirate activity. Pretty standard escort mission with dangerous destination.

Mercenary Prologue 2: FL is called upon to escort the carriers back to New China, with a detour to some small dirt belt mining colony. Upon entering the system with the colony, the convoy is attacked by pirate ships flying FW colors. Arriving back on New China, FL is shuttled up the RM corporate ladder to meet with leadership about this issue.

Mercenary Prologue 3: A patrol mission in the Dirt Belt looking for pirates or ships flying FW colors (with escorts). Orders are search and destroy. Some pirates are found, but all of them are swizzled 6.

Mercenary Prologue 4: When FL lands after the previous mission, they are told that they are urgently needed to track down another "FW" ship that was reported in the dirt belt. Timed bounty mission with NPC support, approaching the scale of a small fleet battle.

Mercenary Prologue 5: Another patrol mission, leading to the discovery of a small pirate outpost on some planet no one cares about. No trace of the green pirates; they seem to have gone to ground.

Mercenary Prologue 6: Raid on the pirate base. The base is only lightly defended, and the battle should be quite easy unless the player behaves very stupidly. This mission involves support from elements of both the FW and the Republic (maybe a Militia Hawk/Navy Gunboat), which helps delay the shooting war. FL is congratulated, and either offered a position in the RM or given a promise of more jobs at a later date depending on what best fits in with other stuff. Nice big bonus.
```

Not sure, but I'm also thinking of maybe keeping some of the same escorts with the player throughout missions 3-6, so that if you take heavy losses early on, the final battle becomes slightly harder (but still not hard), whereas otherwise it is a pushover.

**Reply by SpearDane** [reply]
Sounds great!

**Comment by TheGiraffe3** [comment]
Turned on the "Automatically delete head branches" option (when a PR is merged). Will very slightly increase efficiency, theoretically.
